readthedocs/proxito/tests/test_old_redirects.py

```diff
@@ -0,0 +1,686 @@
+        )
+
+        r = self.client.get(
+            '/en/latest/redirect/',
+            HTTP_HOST='project.dev.readthedocs.io',
+        )
+        self.assertEqual(r.status_code, 404)
+
+
+    def test_redirect_root(self):
+        Redirect.objects.create(
+            project=self.project,
+            redirect_type='prefix',
+            from_url='/woot/',
+        )
+        r = self.client.get(
+            '/woot/faq.html',
+            HTTP_HOST='project.dev.readthedocs.io',
+        )
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'],
+            'http://project.dev.readthedocs.io/en/latest/faq.html',
+        )
+
+    def test_redirect_page(self):
+        Redirect.objects.create(
+            project=self.project,
+            redirect_type='page',
+            from_url='/install.html',
+            to_url='/tutorial/install.html',
+        )
+        r = self.client.get(
+            '/install.html',
+            HTTP_HOST='project.dev.readthedocs.io',
+        )
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'],
+            'http://project.dev.readthedocs.io/en/latest/tutorial/install.html',
+        )
+
+    def test_redirect_with_query_params(self):
+        self._storage_exists([
+            '/media/html/project/latest/tutorial/install.html',
+        ])
+        Redirect.objects.create(
+            project=self.project,
+            redirect_type='page',
+            from_url='/install.html',
+            to_url='/tutorial/install.html',
+        )
+        r = self.client.get(
+            '/install.html?foo=bar',
+            HTTP_HOST='project.dev.readthedocs.io',
+        )
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'],
+            'http://project.dev.readthedocs.io/en/latest/tutorial/install.html?foo=bar',
+        )
+
+    def test_redirect_exact(self):
+        self._storage_exists([
+            '/media/html/project/latest/guides/install.html',
+        ])
+
+        fixture.get(
+            Redirect,
+            project=self.project,
+            redirect_type='exact',
+            from_url='/en/latest/install.html',
+            to_url='/en/latest/tutorial/install.html',
+        )
+        r = self.client.get(
+            '/en/latest/install.html',
+            HTTP_HOST='project.dev.readthedocs.io',
+        )
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'],
+            'http://project.dev.readthedocs.io/en/latest/tutorial/install.html',
+        )
+
+    def test_redirect_exact_with_rest(self):
+        """
+        Exact redirects can have a ``$rest`` in the ``from_url``.
+
+        Use case: we want to deprecate version ``2.0`` and replace it by
+        ``3.0``. We write an exact redirect from ``/en/2.0/$rest`` to
+        ``/en/3.0/``.
+        """
+        self._storage_exists([
```

Comment:
I don't understand why this file needs to exist -- does the redirect only get called when the file _doesn't_ exist?

Reply:
It seems it's wrong and not needed. I should pasted this by mistake while refactoring.